### PR TITLE
add support for weekly invoice

### PR DESCRIPTION
**Pull request by maarten-k**
## Description

Should close invoice: sum calculation on entry per week
#1434

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklistcomposer kimai:code-check
- [x] I verified that my code applies to the guidelines (`composer kimai:code-check`)
- [x] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [x] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

### CODE REVIEW

File: src/Invoice/Calculator/WeeklyInvoiceCalculator.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+use App\Invoice\InvoiceItemInterface;
+
+/**
+ * A calculator that sums up the invoice item records per week.
+ */
+class WeeklyInvoiceCalculator extends AbstractSumInvoiceCalculator implements CalculatorInterface
+{
+    protected function calculateSumIdentifier(InvoiceItemInterface $invoiceItem): string
+    {
+        if (null === $invoiceItem->getUser()->getId()) {
```
Comment (kevinpapst) [inline src/Invoice/Calculator/WeeklyInvoiceCalculator.php:22]: I think this `if` can be removed

**Comment by kevinpapst** [comment]
Thanks @maarten-k - I will give it a try soon

**Comment by maarten-k** [comment]
Documentation can be found at https://github.com/kimai/www.kimai.org/pull/65

File: src/Invoice/Calculator/WeeklyInvoiceCalculator.php (hunk 1 above)
Comment (maarten-k) [reply src/Invoice/Calculator/WeeklyInvoiceCalculator.php]: fixed in b4394da003fd1e9f9ab4654ee6461a9c5b511ffe

**Comment by kevinpapst** [comment]
Thank you @maarten-k 👍

**Pull request merged**